tests/libstat_wrapper.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Simulate FORCESOCKETSFILE existing for the IPC tests
+ */
+
+#include <stdio.h>
+#include <dlfcn.h>
+#include <string.h>
+#include <sys/stat.h>
+#include "../include/config.h"
+#if defined(QB_LINUX) || defined(QB_CYGWIN)
+#include <gnu/lib-names.h>
+#endif
+
+int __xstat(int __ver, const char *__filename, struct stat *__stat_buf)
+{
+#if defined(QB_LINUX) || defined(QB_CYGWIN)
+	static int opened = 0;
+	static void *dlhandle;
+	static int (*real_xstat)(int __ver, const char *__filename, void *__stat_buf);
+
+	if (!opened) {
+		dlhandle = dlopen(LIBC_SO, RTLD_NOW);
+		real_xstat = dlsym(dlhandle, "__xstat");
+		opened = 1;
+	}
+
+	if (strcmp(__filename, FORCESOCKETSFILE) == 0) {
```

Comment:
__xstat seems to return -1 with errno==EFAULT in case the filename points to invalid memory instead of segfaulting.
Don't know if we need it that generic here but we might get called by unexpected callers and get confusing results.